Sources/FrontEnd/Parse/Parser.swift

```diff
@@ -1051,29 +1062,31 @@ public enum Parser {
 
   /// Returns the first memberwise initializer declaration in `members` or synthesizes an implicit
   /// one, appends it into `members`, and returns it.
-  private static func findOrSynthesizeMemberwiseInitDecl(
-    in members: inout [AnyDeclID],
+  private static func getOrCreateMemberwiseInitializer(
+    ofDeclStartingAt startIndex: SourceFile.Index,
+    among members: inout [AnyDeclID],
     updating state: inout ParserState
   ) -> NodeID<InitializerDecl> {
     for member in members where member.kind == InitializerDecl.self {
       let m = NodeID<InitializerDecl>(rawValue: member.rawValue)
       if state.ast[m].introducer.value == .memberwiseInit { return m }
     }
 
+    let site = state.emptyRange(at: startIndex)
```

Comment:
You can do better.  Maybe

```suggestion
    let startOfTypeDecl = state.emptyRange(at: startIndex)
```